lib/jackal-kitchen/tester.rb

```diff
@@ -78,6 +78,18 @@ def execute(msg)
             run_commands(['bundle exec kitchen destroy'], {}, working_dir, payload)
             FileUtils.rm_rf(working_dir)
           end
+
+          failures = payload.get(:data, :kitchen, :test_output, :teapot).any? do |instance, h|
+            h[:run_status][:http_failure][:permanent] == false
+          end
+          retry_count = payload.fetch(:data, :kitchen, :retry_count, 0)
+          retry_count += 1
+
+          payload.set(:data, :kitchen, :retry_count, retry_count) if failures
+
+          cond = failures && retry_count <= app_config.fetch(:kitchen, :config, :retries, 0)
+          payload.set(:data, :kitchen, :retry, cond)
+
```

Comment:
@chrisroberts ah ok cool, thanks. For some reason I was imagining that deleting from the payload was less advisable, but if it's a pattern, shall rock it.